Libs/Visualization/VTK/Core/vtkScalarsToColorsHistogramChart.cpp

```diff
@@ -64,10 +69,62 @@ vtkScalarsToColorsHistogramChart::~vtkScalarsToColorsHistogramChart()
 {
 }
 
+// ----------------------------------------------------------------------------
+void vtkScalarsToColorsHistogramChart::SetLeftAxisMode(int mode)
+{
+  this->LeftAxisMode = mode;
+
+  switch (mode)
+  {
+    case vtkScalarsToColorsHistogramChart::VTK_AUTO:
+      this->GetAxis(vtkAxis::LEFT)->SetBehavior(vtkAxis::AUTO);
+      break;
+    case vtkScalarsToColorsHistogramChart::MAXIMUM:
+    case vtkScalarsToColorsHistogramChart::MEAN_PLUS_THREE_SIGMA:
+      this->GetAxis(vtkAxis::LEFT)->SetBehavior(vtkAxis::FIXED);
+      break;
+    default:
```

Comment:
If mode is not a valid case, it should not be set.

Reply:
not necessarily true if you happen to derive this method and support a new enum value.

Reply:
Good point. In that case, a virtual method `bool isLeftAxisModeValid(int mode) const` could be added.

Reply:
I didn't see your discussion so I just moved the switch above. Just tell me if you want me to improve the new version.